src/libraries/System.Text.Json/Common/JsonNamingPolicy.cs

```diff
@@ -13,6 +13,9 @@ namespace System.Text.Json
 #endif
     abstract class JsonNamingPolicy
     {
+        private const char SnakeWordBoundary = '_';
+        private const char KebabWordBoundary = '-';
```

Comment:
```suggestion
        private const char SnakeCaseSeparator = '_';
        private const char KebabCaseSeparator = '-';
```

Reply:
Moreover, these feel like implementation detail -- might want to move to the implementing class itself and expose factory methods for kebab case and snake case policies.